### PR TITLE
retries for kafka client operations

### PR DESCRIPTION
**Pull request by adyach**
# One-line summary

Retrying on parittions for method of producer / consumer
Log sequence of failed batches to track gaps in publishing

### CODE REVIEW

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
deploy validation please

File: src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java [hunk 1]
```diff
@@ -96,6 +99,66 @@ public KafkaTopicRepository(final ZooKeeperHolder zkFactory,
         this.circuitBreakers = new ConcurrentHashMap<>();
     }
 
+    private CompletableFuture<Exception> publishItem(
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java]: why you moved this method up? Now it's very difficult to see what was actually changed.

File: src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java [hunk 2]
```diff
@@ -401,13 +466,20 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
...
         }
     }
 
     @Override
     public List<String> listPartitionNames(final String topicId) {
+        return Retryer.executeWithRetry(() -> {
+                    return listPartitionNamesInternal(topicId);
+                },
+                new RetryForSpecifiedCountStrategy(3)
+                        .withWaitBetweenEachTry(5000)
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java]: do you think that somebody will wait 15 seconds?

File: src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java (hunk 1 above)
Comment (adyach) [reply src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java]: it was wrongly formatted, and every time I went to this class it changes the sequence of lines, so I decided to commit it.

**Comment by v-stepanov** [comment]
deploy validation please

**Comment by adyach** [comment]
:

**Comment by adyach** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Pull request merged**